packages/taiko-client/prover/init.go

```diff
@@ -218,7 +218,7 @@ func (p *Prover) initEventHandlers() error {
 		ContesterMode:         p.cfg.ContesterMode,
 		ProveUnassignedBlocks: p.cfg.ProveUnassignedBlocks,
 	}
-	if p.IsGuardianProver() {
+	if p.IsGuardianProver() && p.guardianProverHeartbeater != nil {
```

Comment:
If we are running a Guardian prover without a heartbeater, we still need to call `NewBlockProposedEventGuardianHandler`?